### DISCUSSION TITLE
Same user 2 computers, changes lost instead of merged

### DISCUSSION THREAD

**Discussion by janthonyl** [community/community#22917] [New to GitHub]
<p>I am pushing to github from the same Windows Domain and same github account but two different computers (I have a laptop-with-vpn at home, and a Desktop at work).</p>
<p>I seem to be losing my changes. Git seems to think that each submission of the file is a request to overwrite it with a new edition (instead of merging the home-submission’s changes with those of the work-submission).</p>
<p>So if I add 5 lines at work.</p>
<p>Then I go home and add 2 lines at home. </p>
<p>I should have 7 new lines.</p>
<p>Instead I end up with 2  new lines. I’ve lost all my at-work changes.</p>
<p>Suggestions anyone please?</p>
<p>The commands  I use are:</p>
<p>git commit -a -m  “today’s changes”</p>
<p>git fetch origin master</p>
<p>git merge origin/master</p>
<p>git push origin master</p>

**Comment by janthonyl** [comment]
<p>I think I can solve this by having a 2nd github account included in the project as a collaborator.</p>
<p>If anyone  knows a cleaner way to resolve this, please let me know. </p>

**Comment by lee-dohm** [comment, marked as answer]
<p>To test this I:</p>
<ol>
<li>Created <a href="https://github.com/lee-dohm/test-repo/commit/dcd3f82aece6ba607f0690a2f8678bba76a2ae5e" rel="nofollow noopener">this commit</a> creating <code>janthonyl.txt</code> as a blank file using the GitHub web UI</li>
<li>Issued <code>git pull</code> on my local machine with the <code>master</code> branch checked out to put both the GitHub repository and my local repository in the same state</li>
<li>Created <a href="https://github.com/lee-dohm/test-repo/commit/3629cca8dfe38a97c3df005a13515c01b967b2d2" rel="nofollow noopener">this commit</a> adding five lines to the <code>janthonyl.txt</code> file using the GitHub web UI</li>
<li>Added two lines to the local copy of <code>janthonyl.txt</code> using my favorite editor</li>
<li>Issued the command <code>git commit -a -m "Today's changes"</code> locally to commit the local changes</li>
<li>Issued the command <code>git fetch origin master</code> locally to get the changes from the GitHub repository</li>
<li>Issued the command <code>git merge origin/master</code> locally</li>
</ol>
<p>This resulted in the following error message:</p>
<pre><code>Auto-merging janthonyl.txt
CONFLICT (content): Merge conflict in janthonyl.txt
Automatic merge failed; fix conflicts and then commit the result.
</code></pre>
<p>And returned exit code 1, signaling an error.</p>
<p>If you execute <code>git status</code> at this point, you would see:</p>
<pre><code>On branch master
Your branch and 'origin/master' have diverged,
and have 1 and 1 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

You have unmerged paths.
  (fix conflicts and run "git commit")
  (use "git merge --abort" to abort the merge)

Unmerged paths:
  (use "git add &lt;file&gt;..." to mark resolution)

        both modified: janthonyl.txt

no changes added to commit (use "git add" and/or "git commit -a")
</code></pre>
<p>And the way Git is configured by default, executing <code>git push origin master</code> at this point results in this error message:</p>
<pre><code>To https://github.com/lee-dohm/test-repo
 ! [rejected] master -&gt; master (non-fast-forward)
error: failed to push some refs to 'https://github.com/lee-dohm/test-repo'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
</code></pre>
<p>And again, exits with error code 1. Git does all of this specifically <strong>to prevent you from losing data</strong>.</p>
<p>The only way that you could get Git to behave like you’re describing is if instead of <code>git push origin master</code> you executed <code>git push --force origin master</code>, which tells Git that you know what you’re doing and you’re ok with overwriting data. There are probably ways to configure Git to always issue a force push in certain circumstances, but again, this is not how Git behaves by default.</p>
<p>The proper thing to do when you see that merge conflict is to resolve the conflict. You’ll notice that the contents of the file at this point are:</p>
<pre><code>&lt;&lt;&lt;&lt;&lt;&lt;&lt; HEAD
Add two lines at home
2
=======
Add five lines at work
2
3
4
5
&gt;&gt;&gt;&gt;&gt;&gt;&gt; origin/master
</code></pre>
<p>If you want all seven lines, then you could edit the file to look like this:</p>
<pre><code>Add five lines at work
2
3
4
5
Add two lines at home
2
</code></pre>
<p>Then:</p>
<ol>
<li>Execute <code>git add --all</code>
</li>
<li>Execute <code>git commit</code>
</li>
<li>Accept the default merge commit message</li>
<li>Execute <code>git push origin master</code>
</li>
</ol>
<p>Which succeeds, as can be seen in the <a href="https://github.com/lee-dohm/test-repo/commits/master/janthonyl.txt" rel="nofollow noopener">history of the <code>janthonyl.txt</code> file</a> in my test repo.</p>

**Comment by lee-dohm** [comment]
<p>Also, to show that if there are no merge conflicts that things wouldn’t behave the way you’re describing, I created <a href="https://github.com/lee-dohm/test-repo/commits/master/janthonyl2.txt" rel="nofollow noopener">another test with <code>janthonyl2.txt</code></a>. This time I built a file  that had some text that would separate the two changes so that Git’s automatic merge resolution could figure out that the two changes didn’t conflict and handle things for you. So I:</p>
<ol>
<li>Created <code>janthonyl2.txt</code> through the GitHub web UI</li>
<li>
<a href="https://github.com/lee-dohm/test-repo/commit/6f59eb094897485b4d83b2e871992a4a0ddfcc7c#diff-069d6c41e30d6d5fe14cb4c5e08762ba" rel="nofollow noopener">Added some lines to the file</a> to separate the two changes</li>
<li>Executed <code>git pull</code> on my local machine to ensure that the remote and local repositories were in the same state</li>
<li>
<a href="https://github.com/lee-dohm/test-repo/commit/2d8b6230b6cc8ef5c43697a18f7c54c893f29f0a#diff-069d6c41e30d6d5fe14cb4c5e08762ba" rel="nofollow noopener">Added the five lines at the top of the file (before the block)</a> through the GitHub web UI</li>
<li>Added the two lines at the bottom of the file (after the block) locally using my favorite text editor</li>
<li>Executed <code>git commit -a -m "Today's changes"</code> locally</li>
<li>Executed <code>git fetch origin master</code> locally</li>
<li>Executed <code>git merge origin/master</code> locally</li>
</ol>
<p>This time I didn’t get an error, I got the following message:</p>
<pre><code>Auto-merging janthonyl2.txt
Merge made by the 'recursive' strategy.
 janthonyl2.txt | 5 +++++
 1 file changed, 5 insertions(+)
</code></pre>
<p>Notice the text “Merge made by the ‘recursive’ strategy.”, confirming that the merge was automatically resolved. Also, there was no error code returned.</p>
<p>I then was able to execute <code>git push origin master</code> resulting in the file history I linked to at the top of this post.</p>
